### PR TITLE
ENH: Add support to `ma.dot()` for non-2d arrays with `strict=True`

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by mattip** [comment]
Was there a corresponding issue, or was this triggered by noticing the comments in the code?

I am wondering if this has a wider audience fro the fix.

File: numpy/ma/core.py [hunk 1]
```diff
@@ -7827,20 +7842,16 @@ def mask_rowcols(a, axis=None):
...
-        a[:, np.unique(maskedval[1])] = masked
-    return a
+
+    if axis is None:
+        return _mask_propagate(a, axis=(0, 1))
+    elif axis == 0:
+        return _mask_propagate(a, axis=1)
+    else:
+        return _mask_propagate(a, axis=0)
+
```
Comment (MatteoRaso) [inline numpy/ma/core.py:7854]: These changes don't seem to be necessary for the PR. Unless I'm wrong, it'd be best if you left this to a separate PR.

**Review by MatteoRaso — CHANGES_REQUESTED** [review]

**Comment by MatteoRaso** [comment]
>I am wondering if this has a wider audience fro the fix.

I don't have a stake in this, but if Numpy's going to have an ma.dot function, I think it should have the same capabilities as the normal dot function.

**Comment by mattip** [comment]
This deserves a mention in the release notes. See the directions in doc/release/upcoming_changes/README.rst

**Comment by mattip** [comment]
Let's give this a shot. Thanks @F3eQnxN3RriK

**Pull request merged**